From 76f7cd4a284e35e336020e6e058fe2be28dbc0e1 Mon Sep 17 00:00:00 2001
From: Riccardo Magliocchetti <riccardo.magliocchetti@gmail.com>
Date: Sun, 24 Oct 2010 16:26:26 +0200
Subject: [PATCH 04/34] Better check for initrd

Check for /dev/random fails running under systemd. So better
consider checking /init and /linuxrc to know if we are inside
initramfs / initrd.

Should fix bug #13.
---
 bootchartd |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/bootchartd b/bootchartd
index 5da25e4..b85d5e8 100755
--- a/bootchartd
+++ b/bootchartd
@@ -148,7 +148,7 @@ if [ $$ -eq 1 ]; then
 	echo "Starting bootchart logging"
 
 	# Are we running in the initrd ?
-	if [ ! -e /dev/random ]; then
+	if [ -x /init -o -x /linuxrc ]; then
 		IN_INITRD="yes"
 		start &
 	else # running inside the main system
-- 
1.7.3.1

